app/src/main/java/com/tokopedia/sample/utils/UserPrefs.kt

```diff
@@ -0,0 +1,35 @@
+package com.tokopedia.sample.utils
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+/**
+ * Author errysuprayogi on 10,January,2020
+ */
+
+//@SuppressLint("StaticFieldLeak")
+class UserPrefs private constructor(private val context: Context) {
+    private val settings: SharedPreferences
+    fun saveString(key: String?, value: String?) {
+        settings.edit().putString(key, value).commit()
+    }
+
+    fun getString(key: String?, defaultValue: String?): String? {
+        return settings.getString(key, defaultValue)
+    }
+
+    companion object {
+        private var instance: UserPrefs? = null
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/errysuprayogi/TinkerSampleApp/blob/a0b231addeed00eb28338fb98a7d40f48abf7015/app/src/main/java/com/tokopedia/sample/utils/UserPrefs.kt#L22"></span>Do not place Android context classes in static fields (static reference to <code>UserPrefs</code> which has field <code>context</code> pointing to <code>Context</code>); this is a memory leak</td>
    </tr>
  </tbody>
</table>